docs/user/datastores/index_config.rst

```diff
@@ -384,6 +384,52 @@ To set the cardinality of an attribute, use the key ``cardinality`` on the attri
             .addPoint("geom", default = true)
             .build("mySft")
 
+.. _partitioned_indices:
+
+Configuring Partitioned Indices
+-------------------------------
+
+To help with large data sets, GeoMesa can partition each index into separate tables, based on the attributes of
+each feature. Having multiple tables for a single index can make it simpler to manage a cluster, for example by
+making it trivial to delete old data. This functionality is currently supported in HBase, Accumulo and Cassandra.
+
+Partitioning must be specified through user data when creating a simple feature type, before calling
+``createSchema``. To indicate a partitioning scheme, use the key ``geomesa.table.partition``. Currently
+the only valid value is ``time``, to indicate time-based partitioning:
+
+.. tabs::
+
+    .. code-tab:: java
+
+        sft.getUserData().put("geomesa.table.partition", "time");
+
+    .. code-tab:: scala SchemaBuilder
+
+        import org.locationtech.geomesa.utils.geotools.SchemaBuilder
+
+        val sft = SchemaBuilder.builder()
+            .addString("name")
+            .addDate("dtg")
+            .addPoint("geom", default = true)
+            .userData
+            .partitioned()
```

Comment:
To be 'future-proof' do we need to have the partitioned bit take an argument?

Also, I suppose this is a bad idea, but can I partition by a different dtg field than the indexed one?

Reply:
what would the argument be?
the partitioning is SPI loaded, so you could plug in whatever you want. the only one i've provided here is based on the default dtg. the partitioning API passes in the sft, so you could set other flags in the user data to configure a specific partition impl. but i haven't defined that as a common key or anything.

Reply:
the `partitioned` call here is just a convenience, you can always set arbitrary keys using `.userData` instead